src/Microsoft.VisualStudio.Threading/AsyncReaderWriterResourceLock.cs

```diff
@@ -742,7 +742,7 @@ private Task PrepareResourceAsync(TResource resource, CancellationToken cancella
                     using (forConcurrentUse ? this.service.HideLocks() : default(Suppression))
                     {
                         preparationTask = new ResourcePreparationTaskAndValidity(
-                            Task.Factory.StartNew(preparationDelegate, stateObject, CancellationToken.None, TaskCreationOptions.None, TaskScheduler.Default).Unwrap(),
+                            Task.Factory.StartNew(preparationDelegate!, stateObject, CancellationToken.None, TaskCreationOptions.None, TaskScheduler.Default).Unwrap(),
```

Comment:
💭 Does it make sense to move this `!` up to where `preparationDelegate` is initialized, so it's clearer that `forConcurrentUse` is the local variable that identifies the non-null field?

Reply:
This is another case of: it's not that `preparationDelegate` is maybe null, but that it is `Action<object>` where `StartNew` takes an `Action<object?>`.

Reply:
Wrapped in helper.